projects/igniteui-angular/src/lib/directives/drag-drop/drag-drop.directive.ts

```diff
@@ -1275,7 +1275,7 @@ export class IgxDragDirective implements AfterContentInit, OnDestroy {
         // Check for shadowRoot instance and use it if present
         for (const elFromPoint of elementsFromPoint) {
             if (!!elFromPoint?.shadowRoot) {
-                elementsFromPoint = elFromPoint.shadowRoot.elementsFromPoint(pageX, pageY);
+                elementsFromPoint = this.getFromShadowRoot(elFromPoint, pageX, pageY);
```

Comment:
This doesn't look entirely correct, note that the problem existed from the previous version actually.
![image](https://user-images.githubusercontent.com/12942794/172485744-796632da-a83f-40fd-8cbf-837cb9477283.png)

I see 2 problems there:
 a. It's replacing the whole result in each iteration.
 b. It's replacing the same collection in which it's iterating, so it's hard to know if it will yield correct results.

Please make sure to test the fix in Firefox too, because since the previous fix was introduced we have been having a problem with D&D in Firefox.

Reply:
That's under the assumption that the first collection contains a single element, in that case yes the new collection will contain all relevant data, but we can't assume that, right?. The new code looks better. Thanks.